### ISSUE TITLE
gtk::ButtonBuilder is missing icon() and mnemonic()

### ISSUE THREAD

**Issue by HossainAdnan** [gtk-rs/gtk#893]
To achieve the same result as `let add_btn = gtk::Button::new_from_icon_name(Some("list-add"), gtk::IconSize::Button);`, `gtk::ButtonBuilder` must provide `icon()` method, same with mnemonic.

 The `gtk::Button` struct provides 4 constructors: 

- `new()`, 

- `new_with_icon_name(icon_name: Option<&str>, size: IconSize)`, 

- `new_with_label(label: &str)` and 

- `new_with_mnemonic(label: &str)`.

The `ButtonBuilder` misses out on the first and the last one

**Comment by EPashkin** [comment]
Both can't be generated currently as builders works only on properties and GtkButton don't have its.
Icon name need be done manually by creating Image from icon_name and pass it to `image()` (you can add function to `ButtonBuilder` for it if you want)
Seems `use_underline` has same meaning as `with_mnemonic`

**Comment by HossainAdnan** [comment]
> Both can't be generated currently as builders works only on properties and GtkButton don't have its.

Could it be the reason why gtk::SpinButtonBuilder is missing `range()`? I don't see any alternative to the `gtk::SpinButton::new_with_range()` (https://gtk-rs.org/docs/gtk/struct.SpinButton.html#method.new_with_range) in the API (https://gtk-rs.org/docs/gtk/struct.SpinButtonBuilder.html)

**Comment by EPashkin** [comment]
Yes, but GtkSpinButton has ["adjustment" property](https://developer.gnome.org/gtk3/stable/GtkSpinButton.html#GtkSpinButton--adjustment) for that

**Comment by antoyo** [comment]
As a general rule, you need to look at the [properties](https://developer.gnome.org/gtk3/stable/GtkSpinButton.html#GtkSpinButton.properties) of a widget and look for the name you want, but as you can see here, the name can be different.

As `icon-name` is not a property of `gtk::Button` and `new_with_mnemonic()` can be done with `use_underline`, is there anything we need to fix here?
